recipes/mimalloc/all/patches/0003-get-rid-of-mi_install_dir.patch

```diff
@@ -0,0 +1,91 @@
++# set(mi_install_dir "${CMAKE_INSTALL_PREFIX}/lib/mimalloc-${mi_version}")
+ if(MI_SECURE MATCHES "ON")
+   set(mi_basename "mimalloc-secure")
+ else()
+@@ -236,7 +236,7 @@ if(MI_USE_CXX MATCHES "ON")
+ else()
+   message(STATUS "Compiler         : ${CMAKE_C_COMPILER}")
+ endif()
+-message(STATUS "Install directory: ${mi_install_dir}")
++# message(STATUS "Install directory: ${mi_install_dir}")
+ message(STATUS "Build targets    : ${mi_build_targets}")
+ message(STATUS "")
+ 
+@@ -253,7 +253,7 @@ if(MI_BUILD_SHARED)
+   target_link_libraries(mimalloc PUBLIC ${mi_libraries})
+   target_include_directories(mimalloc PUBLIC
+       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+-      $<INSTALL_INTERFACE:${mi_install_dir}/include>
++      # $<INSTALL_INTERFACE:${mi_install_dir}/include>
+   )
+   if(WIN32)
+     # On windows copy the mimalloc redirection dll too.
+@@ -263,8 +263,8 @@ if(MI_BUILD_SHARED)
+       COMMENT "Copy mimalloc-redirect.dll to output directory")
+   endif()
+ 
+-  install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_dir} LIBRARY)
+-  install(EXPORT mimalloc DESTINATION ${mi_install_dir}/cmake)
++  install(TARGETS mimalloc RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
++  # install(EXPORT mimalloc DESTINATION ${mi_install_dir}/cmake)
+ endif()
+ 
+ # static library
+@@ -276,7 +276,7 @@ if (MI_BUILD_STATIC)
+   target_link_libraries(mimalloc-static PUBLIC ${mi_libraries})
+   target_include_directories(mimalloc-static PUBLIC
+       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+-      $<INSTALL_INTERFACE:${mi_install_dir}/include>
++      # $<INSTALL_INTERFACE:${mi_install_dir}/include>
+   )
+   if(WIN32)
+     # When building both static and shared libraries on Windows, a static library should use a
+@@ -287,15 +287,15 @@ if (MI_BUILD_STATIC)
+     set_target_properties(mimalloc-static PROPERTIES OUTPUT_NAME ${mi_basename})
+   endif()
+ 
+-  install(TARGETS mimalloc-static EXPORT mimalloc DESTINATION ${mi_install_dir})
++  install(TARGETS mimalloc-static RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+ endif()
+ 
+ # install include files
+-install(FILES include/mimalloc.h DESTINATION ${mi_install_dir}/include)
+-install(FILES include/mimalloc-override.h DESTINATION ${mi_install_dir}/include)
+-install(FILES include/mimalloc-new-delete.h DESTINATION ${mi_install_dir}/include)
+-install(FILES cmake/mimalloc-config.cmake DESTINATION ${mi_install_dir}/cmake)
+-install(FILES cmake/mimalloc-config-version.cmake DESTINATION ${mi_install_dir}/cmake)
++install(FILES include/mimalloc.h DESTINATION include)
++install(FILES include/mimalloc-override.h DESTINATION include)
++install(FILES include/mimalloc-new-delete.h DESTINATION include)
++# install(FILES cmake/mimalloc-config.cmake DESTINATION ${mi_install_dir}/cmake)
++# install(FILES cmake/mimalloc-config-version.cmake DESTINATION ${mi_install_dir}/cmake)
+ 
+ if(FALSE)
```

Comment:
I see that you have two patches for the same file.
I'm not sure this is a good thing.

I mean, have separate patches for the separate features is good, but applying patches in the right order would be more trouble for the new version of mimalloc.

Reply:
Fixed in 3c836906